vendor/src/github.com/Sirupsen/logrus/writer.go

```diff
@@ -7,18 +7,40 @@ import (
 )
 
 func (logger *Logger) Writer() *io.PipeWriter {
+	return logger.WriterLevel(InfoLevel)
+}
+
+func (logger *Logger) WriterLevel(level Level) *io.PipeWriter {
 	reader, writer := io.Pipe()
 
-	go logger.writerScanner(reader)
+	var printFunc func(args ...interface{})
+	switch level {
```

Comment:
This isn't related to any changes I made. It comes from rebasing off the latest upstream version of logrus.

I can base the changes on the version we had vendored previously if you preferred, but I thought working from the latest upstream would be best.